### DISCUSSION TITLE
Option to pause/stop clients from connecting when using your PC locally

### DISCUSSION THREAD

**Discussion by ardz** [LizardByte/community#337] [Sunshine]
Could this be added to the taskbar icon menu? had a few instances where the family have tapped into my PC from the sheild when I'm working. Love that we have the taskbar icon now (Windows).

![image](https://github.com/LizardByte/Sunshine/assets/6310248/d9540e6f-1f41-4405-95c1-4445ec2fb0b5)

**Comment by ReenigneArcher** [comment]
It's the same as "Quit" ?

**Reply by ardz** [reply]
I guess you could just quit yes, but that means you'd have to start it up all back up. It would be neat if you could just "pause/resume" from the taskbar without having to shut the whole server down.

**Reply by ReenigneArcher** [reply]
It doesn't shut down the server. It just stops the Sunshine service, which seems like what you want to do.

Then you start the service again through the start menu entry.

**Reply by ReenigneArcher** [reply]
Not saying a PR adding this wouldn't be accepted, just providing current steps to achieve the same result.

And if something were implemented, obviously you'd want to reject any new connection attempts... but what if there are existing connections already? Terminate it, or allow it continue?

**Reply by ardz** [reply]
Apologies for the earlier confusion. When I refer to "server," I indeed mean the service that is listening on the ports. :)

![image](https://github.com/LizardByte/Sunshine/assets/6310248/8abc01c8-d4ec-4a5e-8764-96fea28ef63f)

To clarify, yes, an option to reject any incoming connection attempts and to shut down the currently active stream when I'm logged onto the PC locally. Initially, I had thought a keyboard shortcut, such as "WIN-ALT-S," could be used to pause Sunshine streaming. Following this action, a notification could appear saying, "Sunshine streaming paused, press WIN-ALT-S to allow remote connections again," similar to the current "Streaming started" notification. It would be convenient if this keyboard shortcut were configurable.

It takes 10 to 20 seconds for me to restart the service somtimes. Would just be a useful feature for the impatient/lazy like me 😄 

Thanks for considering it all the same. Version 0.21.0 is absolutely brilliant by the way :)

**Reply by ardz** [reply]
> Not saying a PR adding this wouldn't be accepted, just providing current steps to achieve the same result.
> 
> And if something were implemented, obviously you'd want to reject any new connection attempts... but what if there are existing connections already? Terminate it, or allow it continue?

For me, immediately terminating an existsing session is preferable.

**Comment by AsAoA801** [comment]
i need this too, please

**Reply by ReenigneArcher** [reply]
https://github.com/LizardByte/Sunshine/discussions/1831#discussioncomment-7512453

**Comment by OmniiOMEGA** [comment]
> Apologies for the earlier confusion. When I refer to "server," I indeed mean the service that is listening on the ports. :)
> 
> ![image](https://private-user-images.githubusercontent.com/6310248/281540433-8abc01c8-d4ec-4a5e-8764-96fea28ef63f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ1NjY2NDUsIm5iZiI6MTcxNDU2NjM0NSwicGF0aCI6Ii82MzEwMjQ4LzI4MTU0MDQzMy04YWJjMDFjOC1kNGVjLTRhNWUtODc2NC05NmZlYTI4ZWY2M2YucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDFUMTIyNTQ1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGM0OWY2N2EwYmI0ZGI3OTQwOTQ4ZTViYTBkYzY5ODJlMWJkYjJkMTRjMGU5YjgzMzMxMDNjM2NmNTVmNTg3ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.wB9ECu4FqxJY1CR0XE0_NNeckKZ9i20_XzNF6DeL4Ok)
> 
> To clarify, yes, an option to reject any incoming connection attempts and to shut down the currently active stream when I'm logged onto the PC locally. Initially, I had thought a keyboard shortcut, such as "WIN-ALT-S," could be used to pause Sunshine streaming. Following this action, a notification could appear saying, "Sunshine streaming paused, press WIN-ALT-S to allow remote connections again," similar to the current "Streaming started" notification. It would be convenient if this keyboard shortcut were configurable.
> 
> It takes 10 to 20 seconds for me to restart the service somtimes. Would just be a useful feature for the impatient/lazy like me 😄
> 
> Thanks for considering it all the same. Version 0.21.0 is absolutely brilliant by the way :)

Need this feature pls!!
